core/src/main/scala/org/bitcoins/core/psbt/PSBT.scala

```diff
@@ -74,6 +70,14 @@ case class PSBT(
 
   def version: UInt32 = globalMap.version.version
 
+  def validateBIP143Vulnerability: PSBT = {
+    require(isFinalized || inputMaps.size == 1 || inputMaps.forall(
```

Comment:
So isn't the whole point of the vuln is that you cannot verify -- without the full utxo set -- that the PSBT is not paying a large fee? Don't we create a new instance of `PSBT` everytime an action is taken so we can't really tell if we are internally handling PSBT that might be safe, or receiving one from an external data source? 

How are retaining any sort of protection from this vulnerability with this PR?

Reply:
> How are retaining any sort of protection from this vulnerability with this PR?

You are only vulnerable when you are signing so this moves the check to when we sign.  You aren't vulnerable to anything for just having a psbt with a witness utxo.

Reply:
Otherwise this creates annoyances where you have to add things in a certain order. For example, if I already have signature for an input and want to add them and the witness utxo. If I add the witness utxo first, I will get an error because then the invariant will fail, even though it was 100% safe.